controlplane/kubeadm/api/v1alpha3/kubeadm_control_plane_types.go

```diff
@@ -28,6 +28,10 @@ const (
 	KubeadmControlPlaneFinalizer    = "kubeadm.controlplane.cluster.x-k8s.io"
 	KubeadmControlPlaneHashLabelKey = "kubeadm.controlplane.cluster.x-k8s.io/hash"
 
+	// MachineClonedTemplateAnnotation is the machine template annotation that stores json marshalled version of the infrastructure template resource
+	// that was cloned for the machine.
+	MachineClonedTemplateAnnotation = "kubeadm.controlplane.cluster.x-k8s.io/cloned-infrastructure-template"
```

Comment:
For #3020, I think it's important to keep this "source" agnostic – something like `cluster.x-k8s.io/cloned-infrastructure-template` or maybe `cluster.x-k8s.io/cloned-from` if we put the annotation on the cloned infrastructure resource itself as Vince suggested in #3084